app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,436 @@
+        return (c.name.toLowerCase().includes(lowerCasedTerm) || c.display_name.toLowerCase().includes(lowerCasedTerm));
+    });
+};
+
+const filterJoinedChannels = (joinedChannels: MyChannelModel[], allChannels: Channel[] | undefined) => {
+    const ids = joinedChannels.map((c) => c.id);
+    return allChannels?.filter((c) => !ids.includes(c.id));
+};
+
+const makeLeftButton = (icon: ImageResource): OptionsTopBarButton => {
+    return {
+        id: CLOSE_BUTTON_ID,
+        icon,
+        testID: 'close.browse_channels.button',
+    };
+};
+
+const makeRightButton = (theme: Theme, formatMessage: IntlShape['formatMessage']): OptionsTopBarButton => {
+    return {
+        color: theme.sidebarHeaderTextColor,
+        id: CREATE_BUTTON_ID,
+        text: formatMessage({id: 'mobile.create_channel', defaultMessage: 'Create'}),
+        showAsAction: 'always',
+        testID: 'browse_channels.create.button',
+    };
+};
+
+const close = () => {
+    Keyboard.dismiss();
+    dismissModal();
+};
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+        },
+        searchBar: {
+            marginHorizontal: 12,
+            borderRadius: 8,
+            marginTop: 12,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        },
+        searchBarInput: {
+            color: theme.centerChannelColor,
+        },
+        loadingContainer: {
+            flex: 1,
+            justifyContent: 'center' as const,
+            alignItems: 'center' as const,
+        },
+        loading: {
+            height: 32,
+            width: 32,
+            justifyContent: 'center' as const,
+        },
+    };
+});
+
+export default function BrowseChannels({
+    componentId,
+    canCreateChannels,
+    joinedChannels = [],
+    sharedChannelsEnabled,
+    closeButton,
+    currentUserId,
+    currentTeamId,
+    canShowArchivedChannels,
+    categoryId,
+}: Props) {
+    const intl = useIntl();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+
+    const [channels, setChannels] = useState<Channel[]>([]);
+    const [archivedChannels, setArchivedChannels] = useState<Channel[]>([]);
+    const [sharedChannels, setSharedChannels] = useState<Channel[]>([]);
+
+    const [visibleChannels, setVisibleChannels] = useState<Channel[]>([]);
+    const [typeOfChannels, setTypeOfChannels] = useState(PUBLIC);
+
+    const [term, setTerm] = useState('');
+
+    const [adding, setAdding] = useState(false);
+    const [loading, setLoading] = useState(false);
+
+    const mounted = useRef(false);
+    const publicPage = useRef(-1);
+    const sharedPage = useRef(-1);
+    const archivedPage = useRef(-1);
+    const nextPublic = useRef(true);
+    const nextShared = useRef(true);
+    const nextArchived = useRef(true);
+
+    const setHeaderButtons = useCallback((createEnabled: boolean) => {
```

Comment:
why does this need to be memoized?

Reply:
setHeaderButtons is a dependency of onSelectChannel, which is a callback passed to the channel list.